### PR TITLE
PhpStorm inspections

### PR DESCRIPTION
**Pull request by SOF3**
Fixed some minor bugs and dropped some obsolete code

pocketmine\level\generator namespace is ignored in this commit

Also recommended to merge into 0.16 branch

### CODE REVIEW

**Review by PrimusLV — APPROVED** [review]

**Review by dktapps — DISMISSED** [review]
This will break the preprocessor again. Please make corrections.

File: src/pocketmine/inventory/PlayerInventory.php [hunk 1]
```diff
@@ -453,7 +453,7 @@ public function sendSlot($index, $target){
 	 * @return Human|Player
 	 */
 	public function getHolder(){
-		return parent::getHolder();
+		return $this->holder;
```
Comment (SOF3) [inline src/pocketmine/inventory/PlayerInventory.php]: There are already other instances of these overrides. They should not affect performance, and they are indeed necessary for IDEs to identify the type (for autocompletion) in a loosely typed language like PHP.

My change only carries out premature optimization to the method, and does not really change anything that can be worse.

File: src/pocketmine/item/Item.php [hunk 2]
```diff
@@ -589,6 +590,10 @@ public function clearCustomName(){
 		return $this;
 	}
 
+	/**
+	 * @param $name
```
Comment (SOF3) [inline src/pocketmine/item/Item.php]: Probably needs to be changed to `@param string $name`

File: src/pocketmine/metadata/MetadataStore.php [hunk 3]
```diff
@@ -28,7 +28,7 @@
 use pocketmine\utils\PluginException;
 
 abstract class MetadataStore{
-	/** @var \WeakMap[] */
```
Comment (SOF3) [inline src/pocketmine/metadata/MetadataStore.php]: Might want to comment this line out instead.

File: src/pocketmine/utils/MainLogger.php [hunk 4]
```diff
@@ -180,7 +180,7 @@ public function shutdown(){
 	protected function send($message, $level, $prefix, $color){
 		$now = time();
 
-		$thread = \Thread::getCurrentThread();
+		$thread = Thread::getCurrentThread();
```
Comment (SOF3) [inline src/pocketmine/utils/MainLogger.php]: This change is for better IDE autocompletion. You might want to revert this change.

**Review by legoboy0215 — APPROVED** [review]

**Review by remotevase — APPROVED** [review]

**Review by ghost — COMMENTED** [review]
Eh?

**Comment by SOF3** [comment]
Obsoleted by #87
